language/compiler/ir_to_bytecode/src/context.rs

```diff
@@ -289,6 +299,93 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
+    //**********************************************************************************************
+    // Source Maps
+    //**********************************************************************************************
+
+    pub fn add_top_level_function_mapping(
+        &mut self,
+        fdef_idx: FunctionDefinitionIndex,
+        location: Loc,
+    ) -> Result<()> {
+        self.source_map
+            .add_top_level_function_mapping(fdef_idx, location)
+    }
+
+    pub fn add_top_level_struct_mapping(
+        &mut self,
+        struct_def_idx: StructDefinitionIndex,
+        location: Loc,
+    ) -> Result<()> {
+        self.source_map
+            .add_top_level_struct_mapping(struct_def_idx, location)
+    }
+
+    /// Add the source location to the bytecode instruction sequence starting at `offset` within
+    /// the function body given by the `FunctionDefinitionIndex`
+    pub fn add_source_info(
+        &mut self,
+        fdef_idx: FunctionDefinitionIndex,
+        offset: CodeOffset,
+        location: Loc,
+    ) -> Result<()> {
+        self.source_map.add_code_mapping(fdef_idx, offset, location)
+    }
+
+    pub fn add_local_mapping(
+        &mut self,
+        fdef_idx: FunctionDefinitionIndex,
+        variable: Var_,
+    ) -> Result<()> {
+        self.source_map.add_local_mapping(fdef_idx, variable)
+    }
+
+    pub fn add_function_type_parameter_mapping(
+        &mut self,
+        fdef_idx: FunctionDefinitionIndex,
+        ty_var: &TypeVar_,
+    ) -> Result<()> {
+        let source_name = (Identifier::from(ty_var.name()), ty_var.span);
+        self.source_map
+            .add_function_type_parameter_mapping(fdef_idx, source_name)
+    }
+
+    pub fn add_struct_type_parameter_mapping(
+        &mut self,
+        struct_def_idx: StructDefinitionIndex,
+        ty_var: &TypeVar_,
+    ) -> Result<()> {
+        let source_name = (Identifier::from(ty_var.name()), ty_var.span);
+        self.source_map
+            .add_struct_type_parameter_mapping(struct_def_idx, source_name)
+    }
+
+    pub fn add_field_mapping(
+        &mut self,
+        struct_def_idx: StructDefinitionIndex,
+        location: Loc,
+    ) -> Result<()> {
+        self.source_map
+            .add_struct_field_mapping(struct_def_idx, location)
+    }
+
+    pub fn incr_function_index(&mut self) {
```

Comment:
I'd probably prefer a `set` function than an `incr` function

`incr` feels brittle to me